### ISSUE TITLE
Bug on parsing for tests

### ISSUE THREAD

**Issue by lute47lillo** [jorgefandinno/eclingo#40]
Branch addresing it is called lute/bug_tests

```
Given an epistemic rule: ":- &k{ not a}." that is parsed through clingo.

And given the expected program after the epistemic rule has been parsed by clingo: ":- k(not1(u(a))). not1(u(a)) :- not u(a). {k(not1(u(a)))} :- not1(u(a))."

Expected program that is also parsed to compare the AST of the program and the expected program instead of the string representation of them.

```
_Then the test fails._ **Once the AST list of the program and expected program is sorted, they are not equal. Not because they have different AST types within, but because there is an error on the order of the sorting that causes the test to fail.**

I have replicated the failure through a file called bug_lists.py and saved the inputs onto txt files on the folder bug_replication_txt

Parsed.txt is the AST of the parsed program.
Expected.txt is the AST of the expected program.

```
**Epistemic rule -> { k(not1(u(a))) } :- not1(u(a)).**
Expected_rule1.txt is the epistemic rule.
Parsed_rule2.txt is also the epistemic rule.
Epistemic_rule.txt shows that there is no difference between those two txt files except for column position which should not influence the sorting.
```

```
**not1...rule -> not1(u(a)) :- not u(a).**
Expected_rule2.txt is the not1... rule.
Parsed_rule2.txt is also the not1..  rule.
not1_rule.txt shows that there is no difference between those two txt files except for column position which should not influence the sorting.
```

final_difference.txt is the diff command of parsed.txt and expected.txt showing as a whole what I have revealed above.

**Comment by jorgefandinno** [comment]
Please, check  "bug_lists2.py". It simplifies the bug to the following
```
rule1: { k(not1(u(a))) } :- not1(u(a)).
rule2: not1(u(a)) :- not u(a).
rule1 < rule2: True
rule2 < rule1: True
```
Can you try to remove parts of both rules and see if the problem still arises?

**Comment by lute47lillo** [comment]
I'm closing this issue since it is fixed on clingo 5.6.2
